feature/BE/login

dongyeon94 opened the pull request:
(#4 ) login 기능 개발

- 추가된 class
  - **StatusCode**
  HTTP 응답 코드
  - **MemberPK**
  Member 테이블 PK 클래스
  -  **EglooApplication**
  Spring boot Runner 클래스

- 수정내용
  - userId, PK변경
  - Spring Bean 등록 (@RestController, @RequestMapping, @Service)
  - Method 추가 (memberLogin, memberLoginService, findByUserIdAndPassword)

Commit:
(#4) login 기능 개발

upswp commented on back-end-api/src/main/java/com/project/egloo/domain/Member.java:
```diff
@@ -20,8 +21,9 @@
     private UUID id;
-    
-    @ColumnDescription("유저 아이디")
+
+    @Id
+    @ColumnDescription("PK, 유저 아이디")
```
동연님 @Id 두개 설정한 이유가 혹시 있을까요?

dongyeon94 replied:
저희 ERD에 primary키가 userID, ID 두개 잡혀있어서 @id를 추가했습니다.

upswp commented on back-end-api/src/main/java/com/project/egloo/domain/MemberPK.java:
```diff
@@ -0,0 +1,10 @@
+import java.io.Serializable;
+import java.util.UUID;
+
+public class MemberPK implements Serializable {
+    private UUID id;
+    private String userId;
```
이렇게 MemberPK 사용했을때 어떤게 좋을까요?

dongyeon94 replied:
ID가 두개 이상인 복합키 시스템에는 Serializable가 필요하다고 합니다. 구성되지 않을시에는 에러가 발생하게 됩니다.

복합키 매핑 방법으로는
 - @EmbeddedId
 - @IdClass(PK.class)
두가지 방법을 사용하게 된다고 합니다. @EmbeddedId는 결합된 PK가 의미있는 엔티티거나 코드에 재사용될때 사용하게 된다는데 저희는 PK클래스를 따로 코드에 적용할 것 같지는 않아서 IdClass를 적용하게 되었습니다.


* PS
혹시 복합키 적용 방법에 다른 방법이있으면 리뷰 부탁드려요! 😄

upswp reviewed (approved):
PK 부분에서 추가된 부분에서 혹시 참고하신 레퍼런스나 이유 있으시다면 공유 부탁드립니다 ! 이렇게 개발을 해보지 않아서 리뷰 남깁니다 !

upswp commented:
(#4 ) login 기능 개발

저도 이렇게 메세지랑 커밋 메시지 남기도록 할게요 ㅎㅎㅎ !! : )

@haewoniee 이렇게 우리도 남겨요 ! 커밋이랑 PR 날릴때 !!

upswp commented:
@dongyeon94 member관리하는데 복합키를 써야할 이유가 있을까요?

dongyeon94 commented:
@upswp  
ERD와 코드에서 userid를  unique와 not null을 가진 일반 필드로 수정하는 방향은 어떠신가요?
고민을 해보니 다른 테이블과 외래키 관계를 맺을때도 좋지 않을것 같네요

upswp commented:
좋습니다 ! 제가 ERD 수정해두겠습니다 ! 동연님께서 코드 수정해서 커밋해주시고 MR 진행해주시면 좋을것 같습니다 !

dongyeon94 committed:
Update Member.java

@id, @IdClass(MemberPK.class) 어노테이션 삭제, @Column(unique = true), @NotNull,@NotBlank 옵션 추가

dongyeon94 committed:
Delete MemberPK.java

복합키 제거에 따른 불필요한 MemberPK.class 제거

dongyeon94 commented:
커밋 수정완료되었으며 MR진행하겠습니다. ✅
